### PR TITLE
Fix php8.1 compatibility for unit tests part 2

### PR DESCRIPTION
**Pull request by SilinMykola**
### Description (*)
I changed unit tests for PHP 8.1 compatibility

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by andrewbess — APPROVED** [review]

**Review by eduard13 — APPROVED** [review]

File: lib/internal/Magento/Framework/Code/Generator/EntityAbstract.php [hunk 1]
```diff
@@ -157,7 +157,7 @@ public function getSourceClassNameWithoutNamespace()
      */
     protected function _getFullyQualifiedClassName($className)
     {
-        $className = ltrim($className, '\\');
+        $className = $className ? ltrim($className, '\\') : '';
         return $className ? '\\' . $className : '';
```
Comment (Den4ik) [inline lib/internal/Magento/Framework/Code/Generator/EntityAbstract.php]:
```suggestion
        return $className ? '\\' . ltrim($className, '\\') : '';
```

File: lib/internal/Magento/Framework/Mview/Test/Unit/View/ChangelogTest.php [hunk 2]
```diff
@@ -111,6 +111,7 @@ public function testGetNameWithException()
     {
         $this->expectException('DomainException');
         $this->expectExceptionMessage('View\'s identifier is not set');
+        $this->model->setViewId('');
```
Comment (Den4ik) [inline lib/internal/Magento/Framework/Mview/Test/Unit/View/ChangelogTest.php]: Why you we fixed it here but not in `lib/internal/Magento/Framework/Mview/View/Changelog.php`?

File: lib/internal/Magento/Framework/ObjectManager/Config/Config.php [hunk 3]
```diff
@@ -76,18 +76,23 @@ class Config implements \Magento\Framework\ObjectManager\ConfigInterface
...
-    public function __construct(RelationsInterface $relations = null, DefinitionInterface $definitions = null)
-    {
+    public function __construct(
+        RelationsInterface $relations = null,
+        DefinitionInterface $definitions = null,
+        SerializerInterface $serializer = null
+    ) {
         $this->_relations = $relations ?: new \Magento\Framework\ObjectManager\Relations\Runtime();
         $this->_definitions = $definitions ?: new \Magento\Framework\ObjectManager\Definition\Runtime();
+        $this->serializer = $serializer ?: new \Magento\Framework\Serialize\Serializer\Json();
```
Comment (Den4ik) [inline lib/internal/Magento/Framework/ObjectManager/Config/Config.php]: I believe that we should use object manager and `SerializerInterface` for instantiating serializer.

**Review by Den4ik — CHANGES_REQUESTED** [review]
Hi @SilinMykola. Thanks for the work.

Could you please review my comments below.

File: lib/internal/Magento/Framework/ObjectManager/Config/Config.php (hunk 3 above)
Comment (andrewbess) [reply lib/internal/Magento/Framework/ObjectManager/Config/Config.php]: Hello @Den4ik 
We cannot use ObjectManager inside ObjectManager.
You can check it.

File: lib/internal/Magento/Framework/Code/Generator/EntityAbstract.php (hunk 1 above)
Comment (SilinMykola) [reply lib/internal/Magento/Framework/Code/Generator/EntityAbstract.php]: @Den4ik Thanks for your help! I fixed.

File: lib/internal/Magento/Framework/Mview/Test/Unit/View/ChangelogTest.php (hunk 2 above)
Comment (SilinMykola) [reply lib/internal/Magento/Framework/Mview/Test/Unit/View/ChangelogTest.php]: @Den4ik Thanks for your help! I fixed.

File: lib/internal/Magento/Framework/ObjectManager/Config/Config.php (hunk 3 above)
Comment (Den4ik) [reply lib/internal/Magento/Framework/ObjectManager/Config/Config.php]: I'm close this conversation after discussion in Slack.

**Review by andrewbess — APPROVED** [review]

**Comment by Den4ik** [comment]
@magento run all tests

**Review by Den4ik — APPROVED** [review]
Looks ok

**Comment by andrewbess** [comment]
@magento run all tests

**Comment by andrewbess** [comment]
@magento run Unit Tests with env PHP 8.1 with edition CE without extensions magento-commerce/inventory, magento-commerce/adobe-stock-integration, magento-commerce/security-package, magento-commerce/magento2-page-builder, magento-commerce/adobe-ims

File: lib/internal/Magento/Framework/App/Test/Unit/Response/HttpTest.php [hunk 4]
```diff
@@ -245,7 +245,7 @@ public function testSetPublicHeadersWithoutTtl(): void
     {
         $this->expectException('InvalidArgumentException');
         $this->expectExceptionMessage('Time to live is a mandatory parameter for set public headers');
-        $this->model->setPublicHeaders(null);
+        $this->model->setPublicHeaders('');
```
Comment (xmav) [inline lib/internal/Magento/Framework/App/Test/Unit/Response/HttpTest.php]: method argument declared as `int`
https://github.com/magento/magento2/blob/35e4ae66312ecae67508ea066b73e720595b42c1/lib/internal/Magento/Framework/App/Response/Http.php#L128-L132

File: lib/internal/Magento/Framework/ObjectManager/Config/Config.php [hunk 5]
```diff
@@ -335,19 +343,4 @@ public function getPreferences()
...
         return $this->_preferences;
     }
-
-    /**
-     * Get serializer
-     *
-     * @return \Magento\Framework\Serialize\SerializerInterface
-     * @deprecated 101.0.0
-     */
-    private function getSerializer()
```
Comment (xmav) [inline lib/internal/Magento/Framework/ObjectManager/Config/Config.php]: Is there any other reason to move it to constructor except static test failure?
If not - please rollback this change

**Review by xmav — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/App/Test/Unit/Response/HttpTest.php (hunk 4 above)
Comment (SilinMykola) [reply lib/internal/Magento/Framework/App/Test/Unit/Response/HttpTest.php]: Hello @xmav! This fix related to pass null to parameter. I reverted this changes.
Thanks for your CR.

File: lib/internal/Magento/Framework/ObjectManager/Config/Config.php (hunk 5 above)
Comment (SilinMykola) [reply lib/internal/Magento/Framework/ObjectManager/Config/Config.php]: Hi @xmav ! I think some changes reverted before and only this remains. I reverted changes. 
Thanks for your CR.

**Comment by xmav** [comment]
@magento run all tests

**Comment by xmav** [comment]
Unit tests failure is expected and is not reproduced with 2.4-develop
FT B2B failure is not related to changes in this branch

**Comment by xmav** [comment]
@magento imposer code to Magento-trigger/magento2ce

**Comment by m2-github-services** [comment]
@xmav the branch with the code is successfully imported into the `Magento-trigger/magento2ce` repository. Branch name: `imported-SilinMykola-magento2-34450`.

**Pull request merged**